public/app/plugins/datasource/elasticsearch/datasource.ts

```diff
@@ -105,6 +105,15 @@ export class ElasticDatasource extends DataSourceApi<ElasticsearchQuery, Elastic
     }
   }
 
+  private requestAllIndices(indexList: string[], index: number, url: string): Promise<any> {
+    return this.request('GET', indexList[index] + url).catch(err => {
+      if (index === 0) {
+        throw err;
+      }
+      return this.requestAllIndices(indexList, index - 1, url);
```

Comment:
Could we check the error here? It probably doesn't make sense to retry if the response is 500 for example and the server is down.

Reply:
Makes sense! This should only retry when the error code is 404, in this case.